### PR TITLE
Chapter 12: Table Row Deletion, Swipe for Actions, Activity Controller and MVC + challenge

### PR DESCRIPTION
**Pull request by VivianaSara**
###  Changes in code
- in RestaurantTableViewController class I added 2 override functions for swipe-to-left ( includes: delete and share action) & swipe-to-right ( include: mark as favorite action)
- created a new Model class RestaurantDiffableDataSource for activate the swipe action

### UI Updates

- **Delete&Share** 

> swipe-to-left
Screenshot 2022-04-20 at 15 17 35

> after deletion app will looks like ( the cell above is deleted  )

Screenshot 2022-04-20 at 15 18 18

> share action looks like
Screenshot 2022-04-20 at 15 18 34

###  
**    **
- **Mark as favorite**
> swipe-to-right
Screenshot 2022-04-20 at 15 17 42

> after this action the cell will looks like
Screenshot 2022-04-20 at 15 18 07

> remove from favorite 

Screenshot 2022-04-20 at 15 18 02

### CODE REVIEW

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 1]
```diff
@@ -97,8 +94,12 @@ class RestaurantTableViewController: UITableViewController {
 
         let favoriteAction = UIAlertAction(title: nameAction, style: .default, handler: {(_:
                                                                                             UIAlertAction!) -> Void in
-            let cell = tableView.cellForRow(at: indexPath) as? RestaurantTableViewCell
-            cell?.setHeartImage(isFavorite: self.restaurants[indexPath.row].getIsFavorite())
+            guard let cell = tableView.cellForRow(at: indexPath) as? RestaurantTableViewCell
+            else {
```
Comment (ralumarusca) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: For guard statement **else** is usually on the same line with **guard**

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 2]
```diff
@@ -119,6 +120,88 @@ class RestaurantTableViewController: UITableViewController {
         tableView.deselectRow(at: indexPath, animated: false)
     }
 
+    // MARK: - Share and Delete left swipe action
+    override func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt
+                            indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        // Get the selected restaurant
+        guard let restaurant = self.dataSource.itemIdentifier(for: indexPath)
+        else {
+            return UISwipeActionsConfiguration()
```
Comment (ralumarusca) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: here you can also return nil, because the return type of the method is optional

**Review by ralumarusca — CHANGES_REQUESTED** [review]

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 3]
```diff
@@ -119,6 +118,86 @@ class RestaurantTableViewController: UITableViewController {
...
+        // Delete action
+        let deleteAction = UIContextualAction(style: .destructive,
+                                              title: "Delete") { (_, _, completionHandler) in
+            var snapshot = self.dataSource.snapshot()
+            snapshot.deleteItems([restaurant])
+            self.dataSource.apply(snapshot, animatingDifferences: true)
+            // Call completion handler to dismiss the action button
+            completionHandler(true)
+        }
+        // Share action
```
Comment (paulpop10) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift:142]: empty space needed

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 4]
```diff
@@ -119,6 +118,86 @@ class RestaurantTableViewController: UITableViewController {
...
+    // MARK: - Mark as favorite right-swipe action
+    override func tableView(_ tableView: UITableView, leadingSwipeActionsConfigurationForRowAt
+                            indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+
+        let markAsFavoriteAction = UIContextualAction(style: .normal,
+                                                      title: "") { (_, _, completionHandler) in
+            guard let cell = tableView.cellForRow(at: indexPath) as? RestaurantTableViewCell else {
+                return
+            }
+            cell.setHeartImage(isFavorite: self.restaurants[indexPath.row].getIsFavorite())
```
Comment (paulpop10) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift:189]: empty space needed

File: FoodPin/FoodPin/Controller/RestaurantTableViewController.swift [hunk 5]
```diff
@@ -51,15 +48,15 @@ class RestaurantTableViewController: UITableViewController {
...
             tableView: tableView, cellProvider: {[self]tableView, indexPath, restaurant in
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath)
-                    as? RestaurantTableViewCell
-            else {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier,
+                                                           for: indexPath) as? RestaurantTableViewCell else {
                 return UITableViewCell()
             }
+
                 cell.configureCell(nameLabel: restaurant.getName(), locationLabel: restaurant.getLocation(),
```
Comment (paulpop10) [inline FoodPin/FoodPin/Controller/RestaurantTableViewController.swift]: this block needs to be aligned with the rest of the method body. Same for the line underneath it where you set the image

**Review by ralumarusca — APPROVED** [review]

**Review by paulpop10 — APPROVED** [review]

**Pull request merged**